pkg/signature/algorithm.go

```diff
@@ -116,3 +118,13 @@ func Generate(al Algorithm, rdr io.Reader) (public PublicKey, private PrivateKey
 	}
 	return
 }
+
+// Generate a high-level keypair from a given seed
+func GenerateFromSeed(al Algorithm, seed []byte) (public PublicKey, private PrivateKey, err error) {
+	pubBytes, privBytes, err := al.GenerateFromSeed(seed)
```

Comment:
Same question as earlier - is it worth it to do any kind of checking on the seed here? It won't any difference in the outcome, just in the error logged.